addons/account/models/account_account.py

```diff
@@ -654,18 +663,22 @@ def default_get(self, default_fields):
         instead of its name. In that case, switch both fields values.
         """
         if 'name' not in default_fields and 'code' not in default_fields:
-            return super().default_get(default_fields)
-        default_name = self._context.get('default_name')
-        default_code = self._context.get('default_code')
-        if default_name and not default_code:
-            try:
-                default_code = int(default_name)
-            except ValueError:
-                pass
-            if default_code:
-                default_name = False
-        contextual_self = self.with_context(default_name=default_name, default_code=default_code)
-        return super(AccountAccount, contextual_self).default_get(default_fields)
+            res = super().default_get(default_fields)
+        else:
+            default_name = self._context.get('default_name')
+            default_code = self._context.get('default_code')
+            if default_name and not default_code:
+                with contextlib.suppress(ValueError):
+                    default_code = int(default_name)
+                if default_code:
+                    default_name = False
+            contextual_self = self.with_context(default_name=default_name, default_code=default_code)
+            res = super(AccountAccount, contextual_self).default_get(default_fields)
+
+        if 'code_mapping_ids' in default_fields and 'code_mapping_ids' not in res:
+            res['code_mapping_ids'] = [Command.create({'company_id': c.id}) for c in self.env.user.company_ids]
```

Comment:
I don't think that taking all the companies of the users is really what we need
```suggestion
            res['code_mapping_ids'] = [Command.create({'company_id': c.id}) for c in self.env.companies]
```

Reply:
From chatting with Benj, I had the impression that was what we wanted, but I asked him again on Discord just now.